### PR TITLE
Minor UI components fixes

### PR DESCRIPTION
**Pull request by guerrap**
Minor tweaks for the `card` and `accordions` components.

### CODE REVIEW

**Review by Diogomartf — APPROVED** [review]

File: packages/ui/src/components/surfaces/accordion/index.tsx [hunk 1]
```diff
@@ -23,7 +32,7 @@ export interface AccordionProps {
     onChange?: (event: React.MouseEvent, expanded: boolean) => void;
     expanded?: boolean;
     className?: { root?: string };
-    children: ReactElement[];
+    children: ReactNode[];
```
Comment (luzzif) [inline packages/ui/src/components/surfaces/accordion/index.tsx:35]: Does it make sense to have an array here? I think `ReactNode` supports passing stuff like:

```
<div />
<div />
<div />
```

No?

File: packages/ui/src/components/surfaces/accordion/index.tsx [hunk 2]
```diff
@@ -35,14 +44,22 @@ export const Accordion = ({
...
+    );
     const isControlled = useMemo(() => expanded !== undefined, [expanded]);
     const isExpanded = useMemo(
         () => (isControlled ? !!expanded : internalExpanded),
         [isControlled, expanded, internalExpanded]
     );
 
-    const summaryChildren = children[0];
-    const detailsChildren = children[1];
+    const summaryChildren = childSteps.find((child) =>
```
Comment (luzzif) [inline packages/ui/src/components/surfaces/accordion/index.tsx:57]: We've used useMemo above and 2 loops here. Wondering if it makes sense to have a reduce on the children in the `useMemo` to just do one single loop and have both the accordion summary and details.

File: packages/ui/src/components/surfaces/card/index.tsx [hunk 3]
```diff
@@ -41,11 +41,7 @@ export const Card = ({ className, children }: CardProps): ReactElement => {
 
     return (
         <div className={rootStyles({ className: className?.root })}>
-            {titleChildren && (
-                <div className="h-12 cui-flex cui-items-center cui-border-b cui-border-black dark:cui-border-white [&:last-of-type]:cui-border-none">
-                    {titleChildren}
-                </div>
-            )}
+            {titleChildren && titleChildren}
```
Comment (luzzif) [inline packages/ui/src/components/surfaces/card/index.tsx:44]: Can't we just have `titleChildren`?

**Review by luzzif — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/surfaces/accordion/index.tsx (hunk 1 above)
Comment (guerrap) [reply packages/ui/src/components/surfaces/accordion/index.tsx:35]: I meant to change `ReactElement` to `ReactNode` only `Summary` and `Details`, my bad. I'll revert this change.

File: packages/ui/src/components/surfaces/accordion/index.tsx (hunk 2 above)
Comment (guerrap) [reply packages/ui/src/components/surfaces/accordion/index.tsx:57]: Do you mean something like this?
```tsx
const { summaryChildren, detailsChildren } = React.Children.toArray(
    children
).reduce<{
    summaryChildren: ReactNode;
    detailsChildren: ReactNode;
}>(
    (previous, current) => {
        if (matchChildByType(current, AccordionSummary)) {
            return {
                ...previous,
                summaryChildren: current,
            };
        }
        if (matchChildByType(current, AccordionDetails)) {
            return {
                ...previous,
                detailsChildren: current,
            };
        }

        return previous;
    },
    { summaryChildren: null, detailsChildren: null }
);
````

File: packages/ui/src/components/surfaces/accordion/index.tsx [hunk 4]
```diff
@@ -41,8 +50,30 @@ export const Accordion = ({
         [isControlled, expanded, internalExpanded]
     );
 
-    const summaryChildren = children[0];
-    const detailsChildren = children[1];
+    const { summaryChildren, detailsChildren } = React.Children.toArray(
```
Comment (luzzif) [inline packages/ui/src/components/surfaces/accordion/index.tsx:53]: Spreading previous shouldn't be necessary in a reduce, we can just push. What about something like this?

```
const { summaryChildren, detailsChildren } = useMemo(() => {
        return React.Children.toArray(children).reduce(
            (
                accumulator: {
                    summaryChildren: ReactNode[];
                    detailsChildren: ReactNode[];
                },
                child
            ) => {
                if (matchChildByType(child, AccordionSummary))
                    accumulator.summaryChildren.push(child);
                else if (matchChildByType(child, AccordionDetails))
                    accumulator.detailsChildren.push(child);
                return accumulator;
            },
            { summaryChildren: [], detailsChildren: [] }
        );
    }, [children]);
```

**Review by luzzif — CHANGES_REQUESTED** [review]

**Pull request merged**